subsys/bluetooth/audio/has_internal.h

```diff
@@ -0,0 +1,60 @@
+/** @file
+ *  @brief Internal APIs for Bluetooth HAS.
+ */
+
+/*
+ * Copyright (c) 2021 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/types.h>
+#include <bluetooth/bluetooth.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** Hearing Aid Type values */
+#define BT_HAS_HEARING_AID_TYPE_BINAURAL			0x00
+#define BT_HAS_HEARING_AID_TYPE_MONAURAL			0x01
+#define BT_HAS_HEARING_AID_TYPE_BANDED				0x02
+
+/** Hearing Aid Preset Control Point opcodes */
+#define BT_HAS_OP_READ_ALL_PRESETS_REQ				0x00
+#define BT_HAS_OP_READ_PRESET_BY_INDEX_REQ          0x01
+#define BT_HAS_OP_READ_PRESET_RSP             		0x02
+#define BT_HAS_OP_PRESET_CHANGED                  	0x03
+#define BT_HAS_OP_WRITE_PRESET_NAME                 0x04
+#define BT_HAS_OP_SET_ACTIVE_PRESET                 0x05
+#define BT_HAS_OP_SET_NEXT_PRESET               	0x06
+#define BT_HAS_OP_SET_PREV_PRESET               	0x07
+#define BT_HAS_OP_SET_ACTIVE_PRESET_LOCAL_SYNC      0x08
+#define BT_HAS_OP_SET_NEXT_PRESET_LOCAL_SYNC        0x09
+#define BT_HAS_OP_SET_PREV_PRESET_LOCAL_SYNC        0x0a
+
+/** Hearing Access Service Error codes */
+#define BT_HAS_ERR_INVALID_OPCODE					0x80
+#define BT_HAS_ERR_WRITE_NAME_NOT_ALLOWED			0x81
+#define BT_HAS_ERR_SYNC_NOT_SUPPORTED				0x82
+#define BT_HAS_ERR_OPERATION_NOT_POSSIBLE			0x83
+#define BT_HAS_ERR_NAME_TOO_LONG					0x84
+
+struct bt_has {
+	struct bt_has_preset *active_preset;
+	uint8_t features;
+
+	struct bt_gatt_service *service_p;
+};
+
+struct bt_has_features {
+    uint8_t hearing_aid_type:2;
+    uint8_t preset_sync_support:1;
+    uint8_t independent_presets:1;
+    uint8_t dynamic_presets:1;
+    uint8_t writable_presets_support:1;
```

Comment:
Typically we don't use bitfields in the host (not particularly sure what the reason is), but it is recommended to use macros to set/read specific bitfield values instead.